nav2_constrained_smoother/src/constrained_smoother.cpp

```diff
@@ -137,8 +137,8 @@ bool ConstrainedSmoother::smooth(nav_msgs::msg::Path & path, const rclcpp::Durat
       logger_,
       "%s: failed to smooth plan, Ceres could not find a usable solution to optimize.",
       plugin_name_.c_str());
-    throw std::runtime_error(
-            "Failed to smooth plan, Ceres could not find a usable solution.");
+    throw nav2_core::FailedToSmoothPath("Failed to smooth plan, Ceres could not find a usable "
```

Comment:
There aren't any other places where we can use the exceptions in this smoother?

Reply:
Also, no updates to the simple smoother anymore?

Reply:
Or savitsky-golay? These exceptions aren't used anywhere then!

Reply:
We would still have all the exceptions except timeout. 

How about we do this for the savitsky-golay and the simple smoother if and only if all segments fail to be smoothed we throw the FailedToSmoothPath error.

Reply:
Where? In the diff of this PR, there are no modifications to the simple smoother or SG smoother files. The only change to the constrained smoother is the error `FailedToSmoothPath`. Maybe you didnt push something?

Reply:
Nope, 

I just wanted to see if this sounded good to you before proceeding. :).

Reply:
Sounds good!

Reply:
So, 

for 
- https://github.com/ros-planning/navigation2/blob/main/nav2_smoother/src/simple_smoother.cpp#L137
- https://github.com/ros-planning/navigation2/blob/main/nav2_smoother/src/savitzky_golay_smoother.cpp#L70

If we throw a timeout error here, there could be other segments of the path that were smoothed successfully. I thought we wanted to return a partially smoothed path?

Reply:
I think that's true for some things but timeouts are god- when a timeout is exceeded, we return immediately regardless (if you see the code a timeout edits the Impl function. As new segments are called, this is immediately triggered again without any work or very little). So behavior today doesn't actually do anything for additional segments (or nothing of note) when the timeout is exceeded.